### DISCUSSION TITLE
How to calculate KAM from user-defined ebsd document input？

### DISCUSSION THREAD

**Discussion by Ming-is** [mtex-toolbox/mtex#1581] [Ask Anything]
d. txt is the custom ebsd file I use.The ebsd documents I use include xyz and Euler angle. 

I can't identify the grain boundary and calculate kam, and the program always reports errors：Index out of matrix dimension
[d.txt](https://github.com/mtex-toolbox/mtex/files/9715789/d.txt)
The script to get kam is as follows:
% crystal symmetry
CS = crystalSymmetry('6/mmm', [3 3 4.7], 'X||a*', 'Y||b', 'Z||c*', 'mineral', 'Titanium');

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','intoPlane');

%% Specify File Names

% path to files
pname = 'C:\Users\90437\Desktop';

% which files to be imported
fname = [pname '\d.txt'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','generic',...
  'ColumnNames', { 'x' 'y' 'z' 'phi1' 'Phi' 'phi2'}, 'Bunge', 'Radians');

[grains,ebsd.grainId] = calcGrains(ebsd('indexed'));
% remove one-three pixel grains
ebsd(grains(grains.grainSize <= 3)) = [];
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'));

grains = smooth(grains,5);

plot(ebsd('indexed'),ebsd('indexed').orientations)
hold on
plot(grains.boundary,'lineWidth',1.5)
hold off
ebsd = EBSD  
ebsd = ebsd.gridify;
kam = ebsd.KAM / degree;

% lets plot it
plot(ebsd,kam,'micronbar','off')
caxis([0,15])
mtexColorbar
mtexColorMap LaboTeX
hold on
plot(grains.boundary,'lineWidth',1.5)
hold off

what should i do？

**Comment by Tijmenvermeij** [comment]
For which line do you get the error?

**Reply by Ming-is** [reply]
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'));  This line starts to report errors.

**Comment by Ming-is** [comment]
```matlab
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'));
```

```Error reporting starts from the code for grain  division in line 30 ：【Index out of matrix dimension】.
In addition to the above error, the script also has a warning：【Warning: Duplicate data points detected.This may lead to inconsistency between the point and Voronoi area.To avoid this behavior, call UNIQUE on the data point before calling VORONOIN.】.
```
d. txt is the custom ebsd file I use. The following is a screenshot of d.txt
![d](https://user-images.githubusercontent.com/71123582/194992007-8bfb7f03-1fba-4210-8ae8-1840a7a08b93.png)

**Reply by Tijmenvermeij** [reply]
I'm guessing importing your EBSD data goes wrong, I don't know much about that, sorry.
